planner/python_poetry_planner.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2022 Jetpack Technologies Inc and contributors. All rights reserved.
+// Use of this source code is governed by the license in the LICENSE file.
+
+package planner
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/pelletier/go-toml/v2"
+)
+
+type PythonPoetryPlanner struct{}
+
+// PythonPoetryPlanner implements interface Planner (compile-time check)
+var _ Planner = (*PythonPoetryPlanner)(nil)
+
+func (g *PythonPoetryPlanner) Name() string {
+	return "PythonPoetryPlanner"
+}
+
+func (g *PythonPoetryPlanner) IsRelevant(srcDir string) bool {
+	poetryLockPath := filepath.Join(srcDir, "poetry.lock")
+	mainPYPath := filepath.Join(srcDir, "main.py")
+	return fileExists(poetryLockPath) && fileExists(mainPYPath)
+}
+
+func (g *PythonPoetryPlanner) GetPlan(srcDir string) *Plan {
+	version := g.PythonVersion(srcDir)
+	return &Plan{
+		Packages: []string{
+			fmt.Sprintf("python%s", version.majorMinorConcatenated()),
+			"poetry",
+		},
+		InstallStage: &Stage{
+			Command: "poetry install --no-dev --no-interaction --no-ansi",
```

Comment:
Just wondering: does poetry build properly in all scenarios without dev dependencies?

Reply:
`build` int he poetry context is just creating a wheel so it would. But I switched to using pex so no longer building with poetry.